server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java

```diff
@@ -95,22 +95,26 @@ private int calculateGlobalConnections() {
       public Integer apply(EmbeddedCacheManager embeddedCacheManager) {
          CacheManagerJmxRegistration jmxRegistration = SecurityActions.getGlobalComponentRegistry(embeddedCacheManager)
                .getComponent(CacheManagerJmxRegistration.class);
-         try {
-            ObjectName transportNamePattern = new ObjectName(jmxRegistration.getDomain() + ":type=Server,component=Transport,name=*");
-            Set<ObjectName> objectNames = jmxRegistration.getMBeanServer().queryNames(transportNamePattern, null);
-
-            // sum the NumberOfLocalConnections from all transport MBeans that match the pattern
-            int total = 0;
-            for (ObjectName name : objectNames) {
-               if (name.getKeyProperty("name").startsWith(serverName)) {
-                  Integer connections = (Integer) jmxRegistration.getMBeanServer().getAttribute(name, "NumberOfLocalConnections");
-                  total += connections;
+         if (jmxRegistration.enabled()) {
```

Comment:
I'd really love this to be rewritten not to use JMX at some point

Reply:
I've create https://issues.redhat.com/browse/ISPN-11466 to track this.